fail2ban/ipaddr.py

```diff
@@ -163,132 +190,108 @@ def __radd__(self, other):
 		return "%s%s" % (other, self)
 
 	def __hash__(self):
-		return hash(self.addr)^hash((self.plen<<16)|self.family)
+		return hash(self._addr) ^ hash((self._plen << 16) | self._family)
 
+	@property
 	def hexdump(self):
-		""" dump the ip address in as a hex sequence in
-			network byte order - for debug purpose
+		"""Hex representation of the IP address (for debug purposes)
 		"""
 		if self.family == socket.AF_INET:
-			return "%08x" % self.addr
+			return "%08x" % self._addr
 		elif self.family == socket.AF_INET6:
-			return "%032x" % self.addr
+			return "%032x" % self._addr
 		else:
 			return ""
-	
+
+	# TODO: could be lazily evaluated
+	@property
```

Comment:
I really have to get used to it to "fake" methods w/o arguments to be handled like attributes. Especially with this method which is broadly known as a function (`inet_ntoa()`).
But if you prefer it this way, I'm OK with it.